microsoft/validation/yt8m_inference_json_formatter.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2018 Microsoft All Rights Reserved.
+#
+# Active Learning Team
+# ==============================================================================
+
+r"""A simple helper script that formats Audioset Inference CSV files to a norma-
+lized consumable json to be used in model inference result comparisons. The scr-
+pt expects a directory of csv files processed my the Youtube-8m Frame Level mod-
+els and outputs a directory of respected json files converted by this script.
+Usage:
+  # Run a directory of csv files as the input.
+  # Specify a targeted output directory for the json files.
+
+  $ python yt8m_inference_json_formatter.py --csv_files /path/to/csv/files \
+                                    --json_files /path/to/targed/json/file/directory \
+
+  # Future code will include: Model Metrics, Threshold for Audioset Label Confidence
+"""
+from __future__ import print_function
+
+import os
+import csv
+import glob
+import json
+import pandas as pd
+import tensorflow as tf
+from collections import deque
+
+flags = tf.app.flags
+
+flags.DEFINE_string(
+    "csv_files", None, "Path to directory of YT8m csv inference file scores."
+)
+
+flags.DEFINE_string(
+    "json_files", None, "Path to target directory for output json files.")
+
+FLAGS = flags.FLAGS
+
+
+def main(_):
+    # Check to see if flags have been set, otherwise throw ValueError
+    if FLAGS.csv_files:
+        if FLAGS.json_files:
+            # Iterate through each csv file in the directory
+            csv_files = FLAGS.csv_files
+            json_files= FLAGS.json_files
+            for filename in os.listdir(csv_files):
+                if filename.endswith(".csv"):
+                    print("INPUT FILENAME: " + filename)
+                    df = pd.read_csv(filename)
+                    audiosetData = []
+                    # Iterate through each file label data
+                    for index, row in df.iterrows():
+                        # Validate Audiset CSV Inferences
+                        data = {}
+                        rowStr = row['LabelConfidencePairs']
+                        rowLabels = rowStr.split()
+                        data['VideoId'] = row['VideoId']
+                        data['Label_Data'] = {}
+                        # Iterate through each LabelConfidentPair value seperated by space and convert to json record
+                        for newVal in range(0, int(len(rowLabels))): 
+                            # Check to see if the string in the current row is a confidence or a Label index
+                            if newVal % 2 == 0:

```

Comment:
`if newVal % 2 == 0:` is done twice (again on ln 72).
Store the value at the start of `for` loop in something like `is_even = newVal % 2 == 0`

Reply:
[53](https://github.com/evanlouie/fox-audio/blob/nate/audioDev/microsoft/validation/yt8m_inference_json_formatter.py#L53)